tests/core/test_proxy.py

```diff
@@ -44,3 +63,50 @@ def test_no_proxy(self):
         env.pop("https_proxy", None)
         env.pop("HTTP_PROXY", None)
         env.pop("HTTPS_PROXY", None)
+
+class TestProxy(TestCase):
+    @attr(requires='test')
+    def test_proxy(self):
+        config = {
+            "endpoints": {"https://app.datadoghq.com": ["foo"]},
+            "proxy_settings": {
+                "host": "localhost",
+                "port": 3128,
+                "user": None,
+                "password": None
+            }
+        }
+
+        app = Application()
+        app.skip_ssl_validation = True
+        app._agentConfig = config
+
+        trManager = TransactionManager(MAX_WAIT_FOR_REPLAY, MAX_QUEUE_SIZE, THROTTLING_DELAY)
+        trManager._flush_without_ioloop = True  # Use blocking API to emulate tornado ioloop
+        AgentTransaction.set_tr_manager(trManager)
+        app.use_simple_http_client = False # We need proxy capabilities
+        AgentTransaction.set_application(app)
+        AgentTransaction.set_endpoints(config['endpoints'])
+        AgentTransaction._use_blocking_http_client = True # Use the synchronous HTTP client
+
+        AgentTransaction('body', {}, "") # Create and flush the transaction
+        access_log = self.docker_client.exec_start(
+            self.docker_client.exec_create(CONTAINER_NAME, 'cat /var/log/squid/access.log')['Id'])
+        self.assertTrue("CONNECT" in access_log) # There should be an entry in the proxy access log
+        self.assertEquals(len(trManager._endpoints_errors), 1) # There should be an error since we gave a bogus api_key
+
+    def setUp(self):
+        self.docker_client = DockerUtil().client
+
+        for line in self.docker_client.pull(CONTAINER_TO_RUN, stream=True):
+            log.info(line)
+
+        self.container = self.docker_client.create_container(CONTAINER_TO_RUN, detach=True, name=CONTAINER_NAME,
+            ports=[3128], host_config=self.docker_client.create_host_config(port_bindings={3128: 3128}))
+        log.info("Starting container: {0}".format(CONTAINER_TO_RUN))
+        self.docker_client.start(CONTAINER_NAME)
+        sleep(1) # Give time for the container to properly start, otherwise we get 'Proxy CONNECT aborted'
```

Comment:
This is sometimes prone to race conditions. Could we maybe inspect the logs for the `test-squid` container to determine if the process is up or not?